invictus_dashboard/src/app/dashboard/dashboard-task-edit.component.ts

```diff
@@ -0,0 +1,324 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output
+} from '@angular/core';
+import {
+  FormBuilder,
+  FormGroup,
+  AbstractControl,
+  FormArray,
+  Validators
+} from '@angular/forms';
+
+import { DashboardTask, taskPriority } from './dashboard-task';
+import { DashboardService } from './dashboard.service';
+import { NumberValidators } from '../shared/number.validator';
+import { debounceTime } from 'rxjs';
+
+@Component({
+  selector: 'dash-dashboard-task-edit',
+  templateUrl: './dashboard-task-edit.component.html',
+  styleUrls: ['./dashboard-task-edit.component.css'],
+})
+export class DashboardTaskEditComponent implements OnInit, OnDestroy {
+  taskForm: FormGroup;
+  apiErrMsg: string = '';
+  pageTitle: string = 'Add Task';
+  priority = taskPriority;
+  task: DashboardTask;
+  errMsg: { [key: string]: string } = {};
+  taskEditModal!: HTMLElement;
+  sprintEndDateString!: string;
+
+  private validationMsgs = {
+    required: 'This field is required.',
+    range: 'Please enter a number between 1 and 5.',
+  };
+
+  get subTasks(): FormArray {
+    return <FormArray>this.taskForm.get('subTasks');
+  }
+
+  @Input() taskId: number;
+  @Input() sprintEndDate: Date;
+  @Output() taskChangeEvent: EventEmitter<string> = new EventEmitter<string>();
+
+
+  constructor(
+    private fb: FormBuilder,
+    private dashboardService: DashboardService
+  ) {}
+
+  ngOnInit(): void {
+    this.checkComponentInputs();
+    this.buildTaskForm();
+
+    this.addModalEventListeners();
+  }
+
+  addModalEventListeners(): void {
+    this.taskEditModal = document.getElementById('addTaskModal');
```

Comment:
Angular generally doesn't want us doing direct DOM access that uses `document`. There are Angular Directives for that type of manipulation. You should look into ViewChild and ViewChildren. https://angular.io/api/core/ViewChild. It will give you a safe way to access the DOM.

Here's an example of the usage:
```typescript
  @ViewChild('labelFrame', { read: ElementRef })
  labelFrame: ElementRef<HTMLElement>;
```

Reply:
That makes sense - I shifted over the calls I had made to use `@ViewChild` instead.